eng/vmr-sync.sh

```diff
@@ -1,63 +1,47 @@
+### Another usecase is to try manually synchronizing a given commit of some repo into the VMR and
+### trying to Source-Build the VMR. This can help when fixing the Source-Build but using a commit
+### from a not-yet merged branch (or fork) to test the fix will help.
 ###
-### where 'src/aspnetcore/eng/SourceBuild.props' would be the file that is being patched and new
-### changes to it are conflicting with the patch that is being applied.
-###
-### The whole process can be reproduced locally easily by running this script.
-### The patches are located in a folder in the 'dotnet/installer' repository.
-### At the moment of writing, the location is 'src/SourceBuild/patches' but to get
-### the up-to-date location, please see the 'patchesPath' property in
-### https://github.com/dotnet/dotnet/blob/main/src/source-mappings.json
-###
-### You will need to compare what is in the patch and what is in the file and fix the patch.
-###
-### The tooling that synchronizes the VMR will need to clone the various repositories.
-### It clones them into a temporary folder and re-uses them on future runs so it is advised
-### you dedicate a folder to this to speed up your re-runs.
-###
-### This script will synchronize the 'dotnet/dotnet' repo locally and let you inspect the changes.
+### The tooling that synchronizes the VMR will need to clone the various repositories into a temporary
+### folder. These clones can be re-used in future synchronizations so it is advised you dedicate a
+### folder to this to speed up your re-runs.
 ###
 ### USAGE:
-###     ./vmr-sync.sh --tmp-dir "$HOME/repos/tmp"
+###   ./vmr-sync.sh --repository runtime:e7e71da303af8dc97df99b098f21f526398c3943 --tmp-dir "$HOME/repos/tmp"
 ### Options:
+###   --repository name:GIT_REF
+###       Required. Repository + git ref separated by colon to synchronize to.
+###       This can be a specific commit, branch, tag..
 ###   -t, --tmp, --tmp-dir PATH
 ###       Required. Path to the temporary folder where repositories will be cloned
 ###   -v, --vmr, --vmr-dir PATH
 ###       Optional. Path to the dotnet/dotnet repository. When null, gets cloned to the temporary folder
 ###   -b, --branch, --vmr-branch BRANCH_NAME
 ###       Optional. Branch of the 'dotnet/dotnet' repo to synchronize to
-###       This should match the target branch of the PR, defaults to 'main'
-###   --target-ref GIT_REF
-###       Optional. Git ref to synchronize to. This can be a specific commit, branch, tag..
-###       Defaults to the revision of the parent installer repo
-###   --debug
-###       Optional. Turns on the most verbose logging for the VMR tooling
+###       This should match the target branch of the PR; defaults to 'main'
+###   --remote name:URI
+###       Optional. Additional remote to use during the synchronization
+###       This can be used to synchronize to a commit from a fork of the repository
+###       Example: 'runtime:https://github.com/billg/runtime'
+###   --recursive
```

Comment:
My initial thought is that this doesn't feel like the right default behavior in the context of my previous comment.  On the other hand, this could cause some different behavior if specific when using a repo lower in the graph.

Reply:
Yeah, I was worried that a recursive sync from anything else than installer might bring a very wild set of changes.

Do you think it would be worth it to also mention that you should make your changes in the fork based off of the SHA that is in the VMR at the moment? E.g. you'd check the `source-manifest.json`, see that runtime is at `SHA123` and you'd branch off your fix from there?

Reply:
> Do you think it would be worth it to also mention that you should make your changes in the fork based off of the SHA that is in the VMR at the moment

I agree there are scenarios that folks can really mess things up but in general there are a lot of cases where you don't need to work from a fork.  I don't want to warn against those.